packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -73,24 +84,23 @@ def __init__(
         self.processing_strategy = processing_strategy or SequentialProcessing()
 
     @classmethod
-    def from_config(cls, config: dict) -> "DocumentSearch":
+    def from_config(cls, config: DocumentSearchConfig) -> "DocumentSearch":
```

Comment:
isn't there an assumption that it should work on dictionaries or some dict configs ?

Reply:
What do you mean by "assumption"?

Reply:
If you prefer, I can change it so it takes `dict` and then creates the `DocumentSearchConfig` object itself. It's just that I usually prefer specifically typed arguments to untyped arguments, but I can see the argument that `from_config` should be more consistent with `from_config` in other objects

Reply:
Changed - `DocumentSearchConfig.from_config()` now takes dict